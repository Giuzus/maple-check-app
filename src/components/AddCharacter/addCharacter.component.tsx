import { useAtom } from "jotai";
import { useState } from "react";
import { addCharacterAtom } from "../../store/character.store";

function AddCharacter() {
  const [, addCharacter] = useAtom(addCharacterAtom);
  const [character, setCharacter] = useState<string>("");

  return (
    <div className="join">
      <input
        type="text"
        placeholder="Type here"
        className="input input-bordered join-item"
        onChange={(e) => setCharacter(e.target.value)}
        value={character}
      />
      <button
        className="btn join-item"
        onClick={() => {
          if (!character) return;

          addCharacter(character);
          setCharacter("");
        }}
      >
        Add
      </button>
    </div>
  );
}

export default AddCharacter;
