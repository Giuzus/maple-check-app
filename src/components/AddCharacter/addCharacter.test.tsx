import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import AddCharacter from "./addCharacter.component";
import userEvent from "@testing-library/user-event";

describe("Add character", () => {
  test("should match snapshot", () => {
    //Arrange
    const component = render(<AddCharacter />);

    //Assert
    expect(component.baseElement).toMatchSnapshot();
  });

  // test("should add characters", async () => {
  //   //Arrange
  //   const user = userEvent.setup();
  //   render(<AddCharacter />);

  //   //Act
  //   await user.type(screen.getByRole("textbox"), "mock_character");
  //   await user.click(screen.getByText("Add"));

  //   await waitFor(() =>
  //     expect(screen.getByText("mock_character")).toBeTruthy()
  //   );

  //   //Assert
  //   expect(JSON.parse(localStorage.getItem("characters")!)).toEqual([
  //     { name: "mock_character" },
  //   ]);
  // });

  test("should do nothing if input is empty", async () => {
    //Arrange
    const user = userEvent.setup();
    render(<AddCharacter />);

    //Act
    await user.click(screen.getByText("Add"));

    //Assert
    expect(localStorage.getItem("characters")).toBe(null);
  });
});
