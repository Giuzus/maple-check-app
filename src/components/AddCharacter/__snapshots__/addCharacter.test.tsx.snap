// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Add character > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="join"
    >
      <input
        class="input input-bordered join-item"
        placeholder="Type here"
        type="text"
        value=""
      />
      <button
        class="btn join-item"
      >
        Add
      </button>
    </div>
  </div>
</body>
`;
